src/material/datepicker/multi-year-view.ts

```diff
@@ -39,6 +39,43 @@ export const yearsPerPage = 24;
 
 export const yearsPerRow = 4;
 
+/** Gets remainder that is non-negative, even if first number is negative */
+function euclideanModulo (a: number, b: number): number {
```

Comment:
Could you move these helper functions to the bottom of the file? We usually organize them that way